gcf/custom/handler.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package custom
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/datacommonsorg/tools/gcf/lib"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/encoding/prototext"
+)
+
+const (
+	dcManifestPath = "/memfile/core_resolved_mcfs_memfile/core_resolved_mcfs.binarypb"
+)
+
+func HandleTriggerFlow(ctx context.Context, bucket, root string) error {
+	projectID := os.Getenv("projectID")
+	controllerTriggerTopic := os.Getenv("controllerTriggerTopic")
+	if controllerTriggerTopic == "" {
+		return errors.New("controllerTriggerTopic is not set in environment")
+	}
+	reader := GCSReader{}
```

Comment:
Pointer is preferred?

reader := &GCSReader{}

Reply:
Done